test/lib-test.cpp

```diff
@@ -1,13 +1,33 @@
 #include <iostream>
-#include <dotenv.hpp>
+#include <dotenv/dotenv.hpp>
 
 int main(int argc, char **argv) {
-  
-  dotenv::use_dotenv_file("/Users/luclambour/Desktop/Workspace/dotenv-cpp/test/.env");
-  std::cout<<std::getenv("PORT")<<std::endl;
-  std::cout<<std::getenv("COUPLE")<<std::endl;
-  std::cout<<std::getenv("BASE_URL")<<std::endl;
-  std::cout<<std::getenv("KEYSSH")<<std::endl;
+
+  dotenv::use_dotenv_file("/Users/luclambour/workspace/dotenv-cpp/test/.env");
+  std::cout << std::getenv("PORT") << std::endl;
+  std::cout << std::getenv("COUPLE") << std::endl;
+  std::cout << std::getenv("BASE_URL") << std::endl;
+  // std::cout<<std::getenv("KEYSSH")<<std::endl;
+
+  dotenv::getenv_or_value("CRIM1", "Calamity Jane");
+  // Must be Bonnie
+  std::cout << std::getenv("CRIM1") << std::endl;
+  dotenv::getenv_or_value("CRIM3", "Calamity Jane");
+  std::cout << std::getenv("CRIM3") << std::endl;
+
+  std::map<std::string, std::string> far_west{{"CRIM1", "Calamity Jane"},  {"CRIM4", "Bob Dalton"},
+                                              {"CRIM5", "Emmett Dalton"},  {"CRIM6", "William Dalton"},
+                                              {"CRIM7", "Gratton Dalton"}, {"CRIMINAL_FAMILY", "${CRIM4} & ${CRIM5} & ${CRIM6} & ${CRIM7}"}};
+  dotenv::use_dotenv_map(far_west);
```

Comment:
I think the feature we discussed was a bit more complex in the sense that the map is here to define the default values that the application uses from the env while still using .env file value if needed.

```cpp
auto env = dotenv::use_dotenv { .defaults = {
  { "MYVAR", "DEFAULT-VALUE-IF-NOTHING-ON-THE-ENV" }
}}; 

// Then either the app could do :
std::getenv("MYVAR")

//or
env.get("MYVAR");

// Which is way better because it avoids the app from relying on the std::getenv kinda of singleton if the devs wants so.
```

And if we push it a bit further then you could : 
- implement autodocumentation of expected env variable and fail if not available
- implement validation of env variable to have specific types

Regarding specific types you could push it further as such : 

```cpp
struct MyAppEnv {
  int MYVAR{};
  std::optional<std::string> PATH = ".default";
};
BOOST_FUSION_ADAPT_STRUCT(MyAppEnv, MYVAR, PATH);

int main(int argc, char** argv) {
  MyAppEnv env = dotenv::use_dotenv<MyAppEnv> {}; 

  std::cout << env.MYVAR << std::endl;
}
```

If someone would call `export MYVAR="banana"; ./app` you could error out with :
```
Application expected the MYVAR environment variable as an int, provided value "banana" cannot be parsed as an int.
```

This could be implemented using the qi parser and a bit of boost fusion magic.